plugins/inputs/dropwizard/dropwizard.go

```diff
@@ -0,0 +1,350 @@
+}
+
+func (*Dropwizard) Description() string {
+	return "Read Dropwizard-formatted JSON metrics from one or more HTTP endpoints"
+}
+
+func (*Dropwizard) SampleConfig() string {
+	return `
+  ## Works with Dropwizard metrics endpoint out of the box
+
+  ## Multiple URLs from which to read Dropwizard-formatted JSON
+  ## Default is "http://localhost:8081/metrics".
+  urls = [
+    "http://localhost:8081/metrics"
+  ]
+
+  ## Optional SSL Config
+  # ssl_ca = "/etc/telegraf/ca.pem"
+  # ssl_cert = "/etc/telegraf/cert.pem"
+  # ssl_key = "/etc/telegraf/key.pem"
+  ## Use SSL but skip chain & host verification
+  # insecure_skip_verify = false
+
+  ## http request & header timeout
+  timeout = "10s"
+
+  ## exclude some built-in metrics
+  # namedrop = [
+  #  "jvm.classloader*",
+  #  "jvm.buffers*", 
+  #  "jvm.gc*",
+  #  "jvm.memory.heap*",
+  #  "jvm.memory.non-heap*",
+  #  "jvm.memory.pools*",
+  #  "jvm.threads*",
+  #  "jvm.attribute.uptime",
+  #  "jvm.filedescriptor",
+  #  "io.dropwizard.jetty.MutableServletContextHandler*",
+  #  "org.eclipse.jetty.util*" 
+  # ]
+
+  ## include only the required fields (applies to all metrics types)
+  # fieldpass = [
+  #  "count",
+  #  "max",
+  #  "p999",
+  #  "m5_Rate",
+  #  "value" 
+  # ]
+`
+}
+
+func (d *Dropwizard) Gather(acc telegraf.Accumulator) error {
+	if len(d.URLs) == 0 {
+		d.URLs = []string{"http://localhost:8081/metrics"}
+	}
+
+	if d.client == nil {
+		tlsCfg, err := internal.GetTLSConfig(
+			d.SSLCert, d.SSLKey, d.SSLCA, d.InsecureSkipVerify)
+		if err != nil {
+			return err
+		}
+		d.client = &http.Client{
+			Transport: &http.Transport{
+				ResponseHeaderTimeout: d.Timeout.Duration,
+				TLSClientConfig:       tlsCfg,
+			},
+			Timeout: d.Timeout.Duration,
+		}
+	}
+
+	var wg sync.WaitGroup
+	for _, u := range d.URLs {
+		wg.Add(1)
+		go func(url string) {
+			defer wg.Done()
+			if err := d.gatherURL(acc, url); err != nil {
+				acc.AddError(fmt.Errorf("[url=%s]: %s", url, err))
+			}
+		}(u)
+	}
+
+	wg.Wait()
+
+	return nil
+}
+
+// Gauge values can be of different types 
+
+type gaugeValueType int
+
+const (
+        IntType gaugeValueType = iota
+        FloatType
+        StringType
+)
+
+type gaugeValue struct {
+	IntValue int64
+	FloatValue float64
+	StringValue string
+	Type gaugeValueType
+}
+
+func (v *gaugeValue) UnmarshalJSON(b []byte) error {
+	jsonString := string(b)
+	if intValue, err := strconv.ParseInt(jsonString, 10, 64); err == nil {
+		*v = gaugeValue{ 
+			IntValue: intValue,
+			Type: IntType,
+		}
+	} else if floatValue, err := strconv.ParseFloat(jsonString, 64); err == nil {
+		*v = gaugeValue{ 
+			FloatValue: floatValue,
+			Type: FloatType,
+		}
+	} else {
+		*v = gaugeValue{ 
+			StringValue: strings.Trim(jsonString, "\""),
+			Type: StringType,
+		}
+	}
+
+	return nil
+}
+
+type gauge struct {
+	Value gaugeValue `json:"value"`
+}
+
+type counter struct {
+	Count int64 `json:"count"`
+}
+
+type histogram struct {
+	Count  int64   `json:"count"`
```

Comment:
Are these properties intended to be public when the struct is private?

Reply:
I believe that they have to be as "Unmarshal will only set exported fields of the struct.".

Reply:
Fair enough, Unmarshal definitely does require them to be public... It's just strange to see public fields inside private structs. I get used to Golint complaining at me when I do this